src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_TimePicker.cs

```diff
@@ -288,6 +288,47 @@ private async Task When_Native_Flyout_Theme(UIKit.UIUserInterfaceStyle expectedS
 			Assert.AreEqual(expectedStyle, nativeTimePicker.OverrideUserInterfaceStyle);
 		}
 #endif
+#if __IOS__ || __ANDROID__
+		[TestMethod]
+		public async Task When_Time_Uninitialized_Should_Display_Current_Time()
+		{
+			var timePicker = new Microsoft.UI.Xaml.Controls.TimePicker();
+			timePicker.Time = new TimeSpan(-1);
+
+			var expectedCurrentTime = GetCurrentTime();
+
+			TestServices.WindowHelper.WindowContent = timePicker;
+			await TestServices.WindowHelper.WaitForLoaded(timePicker);
+
+			await DateTimePickerHelper.OpenDateTimePicker(timePicker);
+
+			var openFlyouts = FlyoutBase.OpenFlyouts;
+			var associatedFlyout = openFlyouts[0];
+			Assert.IsInstanceOfType(associatedFlyout, typeof(TimePickerFlyout));
+
+#if __ANDROID__
+			var nativeFlyout = (NativeTimePickerFlyout)associatedFlyout;
+			var displayedTime = nativeFlyout._initialTime;
```

Comment:
Instead of relying on the private state of the native flyout, we should be able to actually open the picker and drill down into the native parts and extract the displayed time value from there.